### PR TITLE
implementation of the function ran_pass.

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

**Comment by pfultz2** [comment]
Compile should be changed to call `run_passes`.

File: src/pass_manager.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#include <algorithm>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+void run_passes(program& prog, const std::vector<pass>& passes, tracer trace)
+{
+    if(enabled(MIGRAPHX_TRACE_COMPILE{}))
+        trace = tracer{std::cout};
```
Comment (pfultz2) [inline src/pass_manager.cpp]: I am not sure if we want to move this to compile as the variable is related to compilation, but maybe it make sense to have it enable all passes even when run during quantization.

Comment (scxiao) [reply src/pass_manager.cpp]: Removed enabling the trace here.

**Comment by scxiao** [comment]
> you could probably combine lines 295 and 296 in program.cpp, unless vec_passes will be used sometime in the future

Fixed

**Pull request merged**